src/handler.rs

```diff
@@ -47,28 +49,37 @@ impl RequestHandler {
             }
 
             // Static files
-            match static_files::handle_request(method, headers, root_dir, uri_path, dir_listing)
-                .await
-            {
-                Ok(mut resp) => {
-                    // Append Security Headers
-                    if self.opts.security_headers {
-                        security_headers::with_security_headers(&mut resp);
-                    }
+            let mut response = error_page::get_error_response(method, &http::StatusCode::from_u16(404).unwrap());
+            for try_file in try_files.into_iter() {
+                let search_uri_path = try_file.replace("$uri", uri_path);
+                match static_files::handle_request(method, headers, root_dir, &search_uri_path, dir_listing)
```

Comment:
Unfortunately the code is not the way to go. It has some inconsistencies.

For example there is no guarantee to pass empty values or deformed ones. Also the `search_uri_path` should be an valid URI path format but looks like is just passing the value around based on whatever string that the user can give then passed directly to `handle_request` function, which is wrong.

Other issue is that your code never evaluate for the `--try-files` as optional, the code is just assuming it as required.

Final observation, the possibly `try_files` feature could be better implemented in the `static_files` module as an enhancement for it rather than in the `handler` file over a loop.